docs/src/extend/custom-processors.md

```diff
@@ -139,6 +139,12 @@ See [Specify a Processor](../use/configure/plugins#specify-a-processor) in the P
 
 ## File Extension-named Processor
 
+::: warning
+This feature is deprecated and only works in eslintrc-style configuration files. Flat config files do not automatically apply processors; you need to explicitly set the `processor` property.
+
+**From ESLint v9.0.0+, this feature will be removed.**
```

Comment:
I think that this wording could be confusing because ESLint 9 will still support the eslintrc-style config in legacy mode with `LegacyESLint`. It's the flat config that doesn't support this feature any more.

Reply:
No worries, will do.

I was under the impression that ESLint 9 wouldn't be supporting eslintrc-style _at all_ (except through other tools manually poking at `LegacyESLint` or using `FlatCompat`).

Given that's the impression I've come away with from the docs and announcements (and I like to think I'm a thorough reader!) is there potentially a need to clarify exactly what config system(s) are going to be supported in ESLint 9 and in what ways?

Reply:
The new config system has been in development for years and we've given sporadic updates as we've progressed, and we've reiterated multiple times that v9.0.0 will still support eslintrc. The development plan was laid out in the RFC: https://github.com/eslint/rfcs/tree/main/designs/2019-config-simplification#backwards-compatibility-analysis

Once we have finalized the feature set for v9.0.0, we will also do a larger "What's coming in v9.0.0" blog post where we'll go into more specific details.